src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1192,7 +1209,23 @@ mobile_apis::Result::eType RegisterAppInterfaceRequest::CheckWithPolicyData() {
         message[strings::msg_params][strings::app_hmi_type];
 
     AppHMITypeInserter inserter(app_hmi_type);
-    std::for_each(app_hmi_types.begin(), app_hmi_types.end(), inserter);
+    std::for_each(app_hmi_types_in_policy.begin(),
+                  app_hmi_types_in_policy.end(),
+                  inserter);
+  } else {
+    if (message[strings::msg_params].keyExists(strings::app_hmi_type)) {
```

Comment:
I would extract this if\else to functions, hard to read

Reply:
@LuxoftAKutsan Done in 6ce4904dba86238ab19eb600440ee82f0dd81d37